### ISSUE TITLE
V11.1.3/4 fail to report updates for private repo's specified in .npmrc file where as v11.1.1 did

### ISSUE THREAD

**Issue by GarethBlain** [raineorshine/npm-check-updates#831]
- [x] I have searched for [similar issues](https://github.com/raineorshine/npm-check-updates/issues)
- [x] I am using the latest version of `npm-check-updates` (both 11.1.3 & 11.1.4 have issues. 11.1.2 won't install)
- [x] I am using `node >= 10.17` (V12.16.0)

---------------------------

We have a private registry (Azure npm feed) and v11.1.1 correctly updates from that feed but 11.1.3/4 just ignores all dependencies that are only on that private registry. It updates anything that's on the normal public npm registry fine

## Steps to Reproduce

- Install v11.1.4 of ncu
- Have a .npmrc file pointing to your private registry
- Add a dependency to the package.json that is ONLY on that private registry with a old version num (say "0.0.0")
- Update one of the other dependencies that is on the public npm registry to an old version as well. (say: @angular/animations": "0.0.0",)
- Run ncu

## Current Behavior

ncu will tell you about "@angular/animations" needing updating but not the private registry dependency.

## Expected Behavior

ncu should (like it DOES on v11.1.1) tell you about both the private and public dependencies...

**Comment by raineorshine** [comment]
Thanks for reporting. This is being tracked in #827, so I'll mark this issue as a duplicate.

**Comment by GarethBlain** [comment]
Interestingly tslint is also no longer reported and this is the only deprecated one (not sure if 11.1.1 had a bug that allowed it to see the deprecated versions of tslint) I tried using the --deprecated flag on v11.1.4 and it didn't do anything. tslint STILL doesn't get updated to the latest (deprecated) version and nether do the private registries. Not sure if the  --deprecated flag not working is another bug?

**Comment by raineorshine** [comment]
> Interestingly tslint is also no longer reported and this is the only deprecated one (not sure if 11.1.1 had a bug that allowed it to see the deprecated versions of tslint) I tried using the --deprecated flag on v11.1.4 and it didn't do anything. tslint STILL doesn't get updated to the latest (deprecated) version and nether do the private registries. Not sure if the --deprecated flag not working is another bug?

Thanks, I can reproduce. I'll create a separate issue for this.

**Comment by muzuiget** [comment]
I also have a issue about this, it looks like ncu v11.1.8 don't use the the `registry` in ~/.npmrc, in ncu early version, it used.

I have some config in ~/.npmrc

```
registry=https://registry.npm.taobao.org/
@foobar:registry=https://path/to/the/private/registry
```

When running `ncu -u`, ncu always use the default registry `https://registry.npmjs.org/`, not `https://registry.npm.taobao.org/`. 

I found this because the default registry is rather slow in China, so ncu take too much time to do the check, it random raise the error:

```
FetchError: Request Timeout. npm-registry-fetch defaults to 30000 (30 seconds). Try setting the --timeout option (in milliseconds) to override this.

/home/root/.config/yarn/global/node_modules/npm-check-updates/lib/index.js:377
  throw err
  ^

FetchError: Response timeout while trying to fetch https://registry.npmjs.org/typescript (over 30000ms)
    at Timeout._onTimeout (/home/root/.config/yarn/global/node_modules/minipass-fetch/lib/body.js:128:28)
    at listOnTimeout (internal/timers.js:554:17)
    at processTimers (internal/timers.js:497:7) {
  code: 'FETCH_ERROR',
  errno: 'FETCH_ERROR',
  type: 'body-timeout'
}

``` 

But ncu can get the scoped packages update, mean it use the url in `@foobar:registry`.

**Comment by raineorshine** [comment]
@muzuiget Thank you for the additional information, that's helpful.

This issue is closed as a duplicate of #827. Please post relevant information in #827, or open a new issue for other issues.
